kernel/kernel-api/src/main/java/io/delta/kernel/internal/snapshot/SnapshotHint.java

```diff
@@ -18,17 +18,35 @@
 
 import io.delta.kernel.internal.actions.Metadata;
 import io.delta.kernel.internal.actions.Protocol;
+import java.util.OptionalLong;
 
 /** Contains summary information of a {@link io.delta.kernel.Snapshot}. */
 public class SnapshotHint {
   private final long version;
   private final Protocol protocol;
   private final Metadata metadata;
+  private final OptionalLong tableSizeBytes;
+  private final OptionalLong numFiles;
 
   public SnapshotHint(long version, Protocol protocol, Metadata metadata) {
     this.version = version;
```

Comment:
it would be better to just call `this(...)` and pass in all the args + the two OptionalLong.empty()